src/modules/webpack-publish.test.ts

```diff
@@ -0,0 +1,123 @@
+import { WebpackRestoreOptions } from "../interfaces/webpack-restore-options";
+import { NodeCI } from "./node-ci";
+import { NodeClean } from "./node-clean";
+import { NodeRestore } from "./node-restore";
+import { WebpackPublish } from "./webpack-publish";
+
+describe("webpack-publish", () => {
+    describe("restore", () => {
+        beforeEach(() => {
+            jest.spyOn(NodeClean, "run").mockImplementation();
+            jest.spyOn(NodeClean, "run").mockImplementation();
+            jest.spyOn(NodeCI, "run").mockImplementation();
+        });
+        test("when ci is null, then NodeCI is not called ", () => {
+            // Arrange
+            const nodeCISpy = jest.spyOn(NodeCI, "run");
+
+            const options: WebpackRestoreOptions = {
+                ci: undefined,
+                skipClean: undefined,
+                skipRestore: undefined,
+            };
+
+            // Act
+
+            WebpackPublish.restore(options);
+
+            // Assert
+            expect(nodeCISpy).not.toBeCalled();
+        });
+        test("when ci is false, then NodeCI is not called ", () => {
+            // Arrange
+            const options: WebpackRestoreOptions = {
+                ci: false,
+                skipClean: undefined,
+                skipRestore: undefined,
+            };
+
+            // Act
+            const nodeCISpy = jest.spyOn(NodeCI, "run");
+            WebpackPublish.restore(options);
+
+            // Assert
+            expect(nodeCISpy).not.toBeCalled();
+        });
+        test("when ci is true, then NodeCI is called ", () => {
+            // Arrange
+            const options: WebpackRestoreOptions = {
+                ci: true,
+                skipClean: undefined,
```

Comment:
Question: Is there a reason we need to call out 'skipClean' and 'skipRestore' in the setup in this and other tests when their default is to be `undefined`? Generally if nullable values/default values are not critical to what is being tested, we shouldn't spell them out. It would make reading the tests here easier as what is setup is what is being called out by the test. 

Additionally, @brandongregoryscott while we don't need to add it now, would you see benefit elsewhere in adding Rosie for factories? This is a pretty good usecase for them

Reply:
I actually just added rosie in the PR for the `CommandRegistry` refactor - so we should be all setup to spin up new factories as needed! See `command-definition-factory.ts` as an example. Same approach as on client/other JS repos we have

Reply:
Dang, I needed to pull to get that latest merge. Didn't see it in my local deps. 👍 - then  yeah, should consider using it here